### ISSUE TITLE
file_get_contents() on null in /usr/local/www/nextcloud/lib/private/Files/Filesystem.php:710

### ISSUE THREAD

**Issue by jkoopmann** [rullzer/previewgenerator#145]
Hi,

after uploading thousands of JPGs I get these quite a bit:

`An unhandled exception has been thrown:
Error: Call to a member function file_get_contents() on null in /usr/local/www/nextcloud/lib/private/Files/Filesystem.php:710
Stack trace:
#0 /usr/local/www/nextcloud/lib/private/legacy/image.php(622): OC\Files\Filesystem::file_get_contents('/data/nextcloud...')
#1 /usr/local/www/nextcloud/lib/private/Preview/Image.php(57): OC_Image->loadFromFile('/data/nextcloud...')
#2 /usr/local/www/nextcloud/lib/private/Preview/GeneratorHelper.php(59): OC\Preview\Image->getThumbnail('P1000740.JPG', 4096, 4096, false, Object(OC\Files\View))
#3 /usr/local/www/nextcloud/lib/private/Preview/Generator.php(194): OC\Preview\GeneratorHelper->getThumbnail(Object(OC\Preview\JPEG), Object(OC\Files\Node\File), 4096, 4096)
#4 /usr/local/www/nextcloud/lib/private/Preview/Generator.php(118): OC\Preview\Generator->getMaxPreview(Object(OC\Files\SimpleFS\SimpleFolder), Object(OC\Files\Node\File), 'image/jpeg')
#5 /usr/local/www/nextcloud/lib/private/PreviewManager.php(205): OC\Preview\Generator->getPreview(Object(OC\Files\Node\File), 32, 32, true, 'fill', 'image/jpeg')
#6 /usr/local/www/nextcloud/apps/previewgenerator/lib/Command/PreGenerate.php(205): OC\PreviewManager->getPreview(Object(OC\Files\Node\File), 32, 32, true)
#7 /usr/local/www/nextcloud/apps/previewgenerator/lib/Command/PreGenerate.php(193): OCA\PreviewGenerator\Command\PreGenerate->processFile(Object(OC\Files\Node\File))
#8 /usr/local/www/nextcloud/apps/previewgenerator/lib/Command/PreGenerate.php(164): OCA\PreviewGenerator\Command\PreGenerate->processRow(Array)
#9 /usr/local/www/nextcloud/apps/previewgenerator/lib/Command/PreGenerate.php(132): OCA\PreviewGenerator\Command\PreGenerate->startProcessing()
#10 /usr/local/www/nextcloud/3rdparty/symfony/console/Command/Command.php(251): OCA\PreviewGenerator\Command\PreGenerate->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#11 /usr/local/www/nextcloud/3rdparty/symfony/console/Application.php(946): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#12 /usr/local/www/nextcloud/3rdparty/symfony/console/Application.php(248): Symfony\Component\Console\Application->doRunCommand(Object(OCA\PreviewGenerator\Command\PreGenerate), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#13 /usr/local/www/nextcloud/3rdparty/symfony/console/Application.php(148): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#14 /usr/local/www/nextcloud/lib/private/Console/Application.php(213): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#15 /usr/local/www/nextcloud/console.php(96): OC\Console\Application->run()
#16 /usr/local/www/nextcloud/occ(11): require_once('/usr/local/www/...')
#17 {main}`

Any idea what could be causing this and how to fix it?

Regards,
   JP

**Comment by 4oo4** [comment]
This appears to be a file cache issue. In my logs I see this:

`No cache entry found for /appdata_oc2p2s8mi80a/preview/310260 (storage: local::/var/www/nextcloud/data/, internalPath: appdata_oc2p2s8mi80a/preview/310260)`

I attempted to repair the cache with these, but nothing seemed to work:

```
sudo -u www-data php /var/www/nextcloud/occ files:scan -all
sudo -u www-data php -f /var/www/nextcloud/occ maintenance:mimetype:update-db --repair-filecache
```

Does this plugin have its own cache, or is there another way to rebuild that that I'm not aware of?

**Comment by doits** [comment]
I have the same problem. Here's the error in better formatting:

```
An unhandled exception has been thrown:
Error: Call to a member function file_get_contents() on null in /var/www/owncloud/lib/private/Files/Filesystem.php:711
Stack trace:
#0 /var/www/owncloud/lib/private/legacy/image.php(622): OC\Files\Filesystem::file_get_contents('/var/www/ownclo...')
#1 /var/www/owncloud/lib/private/Preview/Image.php(57): OC_Image->loadFromFile('/var/www/ownclo...')
#2 /var/www/owncloud/lib/private/Preview/GeneratorHelper.php(59): OC\Preview\Image->getThumbnail('DT0A0210.JPG', 4096, 4096, false, Object(OC\Files\View))
#3 /var/www/owncloud/lib/private/Preview/Generator.php(194): OC\Preview\GeneratorHelper->getThumbnail(Object(OC\Preview\JPEG), Object(OC\Files\Node\File), 4096, 4096)
#4 /var/www/owncloud/lib/private/Preview/Generator.php(118): OC\Preview\Generator->getMaxPreview(Object(OC\Files\SimpleFS\SimpleFolder), Object(OC\Files\Node\File), 'image/jpeg')
#5 /var/www/owncloud/lib/private/PreviewManager.php(205): OC\Preview\Generator->getPreview(Object(OC\Files\Node\File), 32, 32, true, 'fill', 'image/jpeg')
#6 /var/www/owncloud/apps/previewgenerator/lib/Command/PreGenerate.php(205): OC\PreviewManager->getPreview(Object(OC\Files\Node\File), 32, 32, true)
#7 /var/www/owncloud/apps/previewgenerator/lib/Command/PreGenerate.php(193): OCA\PreviewGenerator\Command\PreGenerate->processFile(Object(OC\Files\Node\File))
#8 /var/www/owncloud/apps/previewgenerator/lib/Command/PreGenerate.php(164): OCA\PreviewGenerator\Command\PreGenerate->processRow(Array)
#9 /var/www/owncloud/apps/previewgenerator/lib/Command/PreGenerate.php(132): OCA\PreviewGenerator\Command\PreGenerate->startProcessing()
#10 /var/www/owncloud/3rdparty/symfony/console/Command/Command.php(255): OCA\PreviewGenerator\Command\PreGenerate->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#11 /var/www/owncloud/3rdparty/symfony/console/Application.php(901): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#12 /var/www/owncloud/3rdparty/symfony/console/Application.php(262): Symfony\Component\Console\Application->doRunCommand(Object(OCA\PreviewGenerator\Command\PreGenerate), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#13 /var/www/owncloud/3rdparty/symfony/console/Application.php(145): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#14 /var/www/owncloud/lib/private/Console/Application.php(213): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#15 /var/www/owncloud/console.php(97): OC\Console\Application->run()
#16 /var/www/owncloud/occ(11): require_once('/var/www/ownclo...')
#17 {main}#
```

The file is changes after each run, eg. in the above error it is `DT0A0210.JPG`, on next run it is `DT0A0211.JPG` etc., so it seems it skips the offending file.

**Comment by jkoopmann** [comment]
Just switched to NC16 and as such cannot test at the moment (or is the plugin already available for NC16?).

**Comment by pousterlus** [comment]
Hi,

Same thing here. I tried to Google that and find a solution but nothing worked (remove appdata files, occ:maintenance repair and other occ's stuff).
In the NC logs, nothing shows up, only the stacktrace when generating-all thumbnails :

```
An unhandled exception has been thrown:
Error: Call to a member function file_get_contents() on null in /usr/local/www/nextcloud/lib/private/Files/Filesystem.php:711
Stack trace:
#0 /usr/local/www/nextcloud/lib/private/legacy/image.php(622): OC\Files\Filesystem::file_get_contents('/usr/local/www/...')
#1 /usr/local/www/nextcloud/lib/private/Preview/Image.php(57): OC_Image->loadFromFile('/usr/local/www/...')
#2 /usr/local/www/nextcloud/lib/private/Preview/GeneratorHelper.php(59): OC\Preview\Image->getThumbnail('2043.JPG', 4096, 4096, false, Object(OC\Files\View))
#3 /usr/local/www/nextcloud/lib/private/Preview/Generator.php(194): OC\Preview\GeneratorHelper->getThumbnail(Object(OC\Preview\JPEG), Object(OC\Files\Node\File), 4096, 4096)
#4 /usr/local/www/nextcloud/lib/private/Preview/Generator.php(118): OC\Preview\Generator->getMaxPreview(Object(OC\Files\SimpleFS\SimpleFolder), Object(OC\Files\Node\File), 'image/jpeg')
#5 /usr/local/www/nextcloud/lib/private/PreviewManager.php(205): OC\Preview\Generator->getPreview(Object(OC\Files\Node\File), 32, 32, true, 'fill', 'image/jpeg')
#6 /usr/local/www/nextcloud/apps/previewgenerator/lib/Command/PreGenerate.php(205): OC\PreviewManager->getPreview(Object(OC\Files\Node\File), 32, 32, true)
#7 /usr/local/www/nextcloud/apps/previewgenerator/lib/Command/PreGenerate.php(193): OCA\PreviewGenerator\Command\PreGenerate->processFile(Object(OC\Files\Node\File))
#8 /usr/local/www/nextcloud/apps/previewgenerator/lib/Command/PreGenerate.php(164): OCA\PreviewGenerator\Command\PreGenerate->processRow(Array)
#9 /usr/local/www/nextcloud/apps/previewgenerator/lib/Command/PreGenerate.php(132): OCA\PreviewGenerator\Command\PreGenerate->startProcessing()
#10 /usr/local/www/nextcloud/3rdparty/symfony/console/Command/Command.php(255): OCA\PreviewGenerator\Command\PreGenerate->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#11 /usr/local/www/nextcloud/3rdparty/symfony/console/Application.php(901): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#12 /usr/local/www/nextcloud/3rdparty/symfony/console/Application.php(262): Symfony\Component\Console\Application->doRunCommand(Object(OCA\PreviewGenerator\Command\PreGenerate), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#13 /usr/local/www/nextcloud/3rdparty/symfony/console/Application.php(145): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#14 /usr/local/www/nextcloud/lib/private/Console/Application.php(213): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#15 /usr/local/www/nextcloud/console.php(97): OC\Console\Application->run()
#16 /usr/local/www/nextcloud/occ(11): require_once('/usr/local/www/...')
```
Can someone point me to a good direction ?
Thanks,

-- 
Leo.

**Comment by mhbates** [comment]
I'm having a similar issue. I'm using Docker, hence the difference in /data location.

`An unhandled exception has been thrown:
Error: Call to a member function file_get_contents() on null in /nextcloud/lib/private/Files/Filesystem.php:711
Stack trace:
#0 /nextcloud/lib/private/legacy/image.php(622): OC\Files\Filesystem::file_get_contents('/data/user/file...')
#1 /nextcloud/lib/private/Preview/Image.php(57): OC_Image->loadFromFile('/data/user/file...')
#2 /nextcloud/lib/private/Preview/GeneratorHelper.php(59): OC\Preview\Image->getThumbnail('Messenger.arura...', 4096, 4096, false, Object(OC\Files\View))
#3 /nextcloud/lib/private/Preview/Generator.php(194): OC\Preview\GeneratorHelper->getThumbnail(Object(OC\Preview\PNG), Object(OC\Files\Node\File), 4096, 4096)
#4 /nextcloud/lib/private/Preview/Generator.php(118): OC\Preview\Generator->getMaxPreview(Object(OC\Files\SimpleFS\SimpleFolder), Object(OC\Files\Node\File), 'image/png')
#5 /nextcloud/lib/private/PreviewManager.php(205): OC\Preview\Generator->getPreview(Object(OC\Files\Node\File), 32, 32, true, 'fill', 'image/png')
#6 /apps2/previewgenerator/lib/Command/PreGenerate.php(205): OC\PreviewManager->getPreview(Object(OC\Files\Node\File), 32, 32, true)
#7 /apps2/previewgenerator/lib/Command/PreGenerate.php(193): OCA\PreviewGenerator\Command\PreGenerate->processFile(Object(OC\Files\Node\File))
#8 /apps2/previewgenerator/lib/Command/PreGenerate.php(164): OCA\PreviewGenerator\Command\PreGenerate->processRow(Array)
#9 /apps2/previewgenerator/lib/Command/PreGenerate.php(132): OCA\PreviewGenerator\Command\PreGenerate->startProcessing()
#10 /nextcloud/3rdparty/symfony/console/Command/Command.php(255): OCA\PreviewGenerator\Command\PreGenerate->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#11 /nextcloud/3rdparty/symfony/console/Application.php(901): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#12 /nextcloud/3rdparty/symfony/console/Application.php(262): Symfony\Component\Console\Application->doRunCommand(Object(OCA\PreviewGenerator\Command\PreGenerate), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#13 /nextcloud/3rdparty/symfony/console/Application.php(145): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#14 /nextcloud/lib/private/Console/Application.php(213): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#15 /nextcloud/console.php(97): OC\Console\Application->run()
#16 /nextcloud/occ(11): require_once('/nextcloud/cons...')
#17 {main}`

**Comment by mhbates** [comment]
Quick update after having tried many different things, like clearing appdata, scanning, redoing initial preview gen, etc. After this second redo of the initial generation, I saw that the recurring task generation had the same error as usual (see my last post). I then did another appdata scan, which took around 5 minutes, and after that the recurring generation did not throw an error. So either the appdata scan after initial gen fixed it, or there were only a few problem files and the generator happened to finish "ignoring" them after this last appdata scan.
